### ISSUE TITLE
Error: Dependency package(s) 'SparkR' not available

### ISSUE THREAD

**Issue by heuermh** [bigdatagenomics/adam#2086]
R build complains about not being able to find SparkR package
```
$ mvn -Pr package
...
[INFO] --- exec-maven-plugin:1.5.0:exec (doc-r) @ adam-r-spark2_2.11 ---

R version 3.5.1 (2018-07-02) -- "Feather Spray"
Copyright (C) 2018 The R Foundation for Statistical Computing
Platform: x86_64-apple-darwin16.7.0 (64-bit)

R is free software and comes with ABSOLUTELY NO WARRANTY.
You are welcome to redistribute it under certain conditions.
Type 'license()' or 'licence()' for distribution details.

  Natural language support but running in an English locale

R is a collaborative project with many contributors.
Type 'contributors()' for more information and
'citation()' on how to cite R or R packages in publications.

Type 'demo()' for some demos, 'help()' for on-line help, or
'help.start()' for an HTML browser interface to help.
Type 'q()' to quit R.

> library(devtools);devtools::document()
Updating bdgenomics.adam documentation
Updating roxygen version in adam/adam-r/bdgenomics.adam/DESCRIPTION
Writing NAMESPACE
Loading bdgenomics.adam
Error: Dependency package(s) 'SparkR' not available.
In addition: Warning messages:
1: roxygen2 requires Encoding: UTF-8
2: In (function (dep_name, dep_ver = "*")  :
  Dependency package 'SparkR' not available.
Execution halted
[ERROR] Command execution failed.
org.apache.commons.exec.ExecuteException: Process exited with an error: 1 (Exit value: 1)
```

I thought I had the SparkR package installed already, tried again, same result
```bash
SparkR
$ cd $SPARK_HOME/R/lib/SparkR/

SparkR
$ R -e "devtools::install('.')"

R version 3.5.1 (2018-07-02) -- "Feather Spray"
Copyright (C) 2018 The R Foundation for Statistical Computing
Platform: x86_64-apple-darwin16.7.0 (64-bit)

R is free software and comes with ABSOLUTELY NO WARRANTY.
You are welcome to redistribute it under certain conditions.
Type 'license()' or 'licence()' for distribution details.

  Natural language support but running in an English locale

R is a collaborative project with many contributors.
Type 'contributors()' for more information and
'citation()' on how to cite R or R packages in publications.

Type 'demo()' for some demos, 'help()' for on-line help, or
'help.start()' for an HTML browser interface to help.
Type 'q()' to quit R.

> devtools::install('.')
   checking for file ‘/usr/local/Cellar/apache-spark/2.3.2/libexec/R/lib/SparkR/✔  checking for file ‘/usr/local/Cellar/apache-spark/2.3.2/libexec/R/lib/SparkR/DESCRIPTION’
─  preparing ‘SparkR’:
✔  checking DESCRIPTION meta-information ...
   Warning in file(con, "r") :
     cannot open file 'man': No such file or directory
    ERROR
   computing Rd index failed:cannot open the connection
Error in run(bin, args = real_cmdargs, stdout_line_callback = real_callback(stdout),  :
  System command error
Calls: <Anonymous> ... with.default -> eval -> eval -> with_envvar -> force -> run
Execution halted

SparkR
$ mkdir man

SparkR
$ R -e "devtools::install('.')"

R version 3.5.1 (2018-07-02) -- "Feather Spray"
Copyright (C) 2018 The R Foundation for Statistical Computing
Platform: x86_64-apple-darwin16.7.0 (64-bit)

R is free software and comes with ABSOLUTELY NO WARRANTY.
You are welcome to redistribute it under certain conditions.
Type 'license()' or 'licence()' for distribution details.

  Natural language support but running in an English locale

R is a collaborative project with many contributors.
Type 'contributors()' for more information and
'citation()' on how to cite R or R packages in publications.

Type 'demo()' for some demos, 'help()' for on-line help, or
'help.start()' for an HTML browser interface to help.
Type 'q()' to quit R.

> devtools::install('.')
   checking for file ‘/usr/local/Cellar/apache-spark/2.3.2/libexec/R/lib/SparkR/✔  checking for file ‘/usr/local/Cellar/apache-spark/2.3.2/libexec/R/lib/SparkR/DESCRIPTION’
─  preparing ‘SparkR’:
✔  checking DESCRIPTION meta-information ...
─  checking whether ‘INDEX’ is up-to-date ... NO
─  use ‘--force’ to remove the existing ‘INDEX’
─  excluding invalid files
   Subdirectory 'R' contains invalid file names:
     ‘SparkR’ ‘SparkR.rdb’ ‘SparkR.rdx’
─  checking for LF line-endings in source and make files and shell scripts
─  checking for empty or unneeded directories
   Removed empty directory ‘SparkR/R’
   Removed empty directory ‘SparkR/man’
   WARNING: Removing directory ‘SparkR/Meta’ which should only occur in an
     installed package
   WARNING: Removing directory ‘SparkR/help’ which should only occur in an
     installed package
   WARNING: Removing directory ‘SparkR/html’ which should only occur in an
     installed package
─  building ‘SparkR_2.3.2.tar.gz’

Running /usr/local/Cellar/r/3.5.1/lib/R/bin/R CMD INSTALL \
  /var/folders/m6/4yqn_4q129lbth_dq3qzj_8h0000gn/T//RtmpkLFoFF/SparkR_2.3.2.tar.gz \
  --install-tests
* installing to library ‘/usr/local/lib/R/3.5/site-library’
* installing *binary* package ‘SparkR’ ...
* DONE (SparkR)

$ mvn -Pr package
...
[INFO] --- exec-maven-plugin:1.5.0:exec (doc-r) @ adam-r-spark2_2.11 ---

R version 3.5.1 (2018-07-02) -- "Feather Spray"
Copyright (C) 2018 The R Foundation for Statistical Computing
Platform: x86_64-apple-darwin16.7.0 (64-bit)

R is free software and comes with ABSOLUTELY NO WARRANTY.
You are welcome to redistribute it under certain conditions.
Type 'license()' or 'licence()' for distribution details.

  Natural language support but running in an English locale

R is a collaborative project with many contributors.
Type 'contributors()' for more information and
'citation()' on how to cite R or R packages in publications.

Type 'demo()' for some demos, 'help()' for on-line help, or
'help.start()' for an HTML browser interface to help.
Type 'q()' to quit R.

> library(devtools);devtools::document()
Updating bdgenomics.adam documentation
Loading bdgenomics.adam
Error: Dependency package(s) 'SparkR' not available.
In addition: Warning messages:
1: roxygen2 requires Encoding: UTF-8
2: In (function (dep_name, dep_ver = "*")  :
  Dependency package 'SparkR' not available.
Execution halted
[ERROR] Command execution failed.
org.apache.commons.exec.ExecuteException: Process exited with an error: 1 (Exit value: 1)
```

**Comment by heuermh** [comment]
Got it, needed to re-export `SPARKR_SUBMIT_ARGS`
```
...
* checking tests ...
  Running ‘testthat.R’
 OK
* checking PDF version of manual ... OK
* DONE

Status: 1 WARNING, 2 NOTEs
See
  ‘adam/adam-r/bdgenomics.adam.Rcheck/00check.log’
for details.


[INFO]
[INFO] --- maven-compiler-plugin:3.8.0:testCompile (default-testCompile) @ adam-r-spark2_2.11 ---
[INFO] No sources to compile
[INFO]
[INFO] --- maven-surefire-plugin:2.22.1:test (default-test) @ adam-r-spark2_2.11 ---
[INFO] No tests to run.
[INFO]
[INFO] --- maven-jar-plugin:3.1.0:jar (default-jar) @ adam-r-spark2_2.11 ---
[WARNING] JAR will be empty - no content was marked for inclusion!
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  01:04 min
[INFO] Finished at: 2018-11-13T16:31:20-06:00
[INFO] ------------------------------------------------------------------------

adam-r (HEAD detached at adam-parent-spark2_2.11-0.25.0)!
$ R CMD build bdgenomics.adam
* checking for file ‘bdgenomics.adam/DESCRIPTION’ ... OK
* preparing ‘bdgenomics.adam’:
* checking DESCRIPTION meta-information ... OK
* checking for LF line-endings in source and make files and shell scripts
* checking for empty or unneeded directories
* building ‘bdgenomics.adam_0.25.0.tar.gz’
Warning in utils::tar(filepath, pkgname, compression = "gzip", compression_level = 9L,  :
  storing paths of more than 100 bytes is not portable:
  ‘bdgenomics.adam/man/rightOuterBroadcastRegionJoinAndGroupByRight-GenomicDataset-GenomicDataset-method.Rd’
Warning in utils::tar(filepath, pkgname, compression = "gzip", compression_level = 9L,  :
  storing paths of more than 100 bytes is not portable:
  ‘bdgenomics.adam/man/rightOuterShuffleRegionJoinAndGroupByLeft-GenomicDataset-GenomicDataset-method.Rd’
```

**Comment by heuermh** [comment]
Fixed by #2090
